### ISSUE TITLE
Allow all 12.x and 14.x Node.js versions to be used

### ISSUE THREAD

**Issue by lirantal** [typescript-eslint/typescript-eslint#4465]
Since this commit https://github.com/typescript-eslint/typescript-eslint/commit/b63013adf0239d98ca9368d73fda057f3b0e57f9 and specifically the major version 5.x of the plugin, this package is requiring specifically the v14.17 version of Node.js as part of the `engines` rule, such as:

```
  "engines": {
    "node": "^12.22.0 || ^14.17.0 || >=16.0.0"
  },
```

Is there a reason for specifically setting a lower cap on v12.22 and v14.17 minor versions? Because if this is about being LTS compliant than the rule should really be:

```
  "engines": {
    "node": "^12.0.0 || ^14.0.0 || >=16.0.0"
  },
```

To allow all supported (v12 and v14) Node.js versions to be used.

## Why didn't we hear about this before?

This might not seem as an issue for npm users, because the default engines configuration is to shoot out a WARNING issue, rather than failing it, but some versions of yarn, such as 1.22.10 explicitly fails any `npm install` of such projects. See here:

```
$ yarn
yarn install v1.22.10
warning package-lock.json found. Your project contains lock files generated by tools other than Yarn. It is advised not to mix package managers in order to avoid resolution inconsistencies caused by unsynchronized lock files. To clear this warning, remove package-lock.json.
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
error @typescript-eslint/eslint-plugin@5.10.0: The engine "node" is incompatible with this module. Expected version "^12.22.0 || ^14.17.0 || >=16.0.0". Got "14.15.1"
error Found incompatible module.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
```

If you confirm this issue and would be open to receiving a fix to relax the Node.js engines config on package.json then I'm happy to push forward a quick PR to suggest the fix/

- [ ] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

**Repro**

1. Run `npm init -y`
2. Use Node.js version v16 (any patch and any minor)
3. Run `npm install` 
4. All work fine

1. Run `npm init -y`
2. Use Node.js version v14.15.1
3. Update the `package.json` to include:
```
  "engines": {
    "node": "^12.22.0 || ^14.17.0 || >=16.0.0"
  },
```
4. Run `npm install`
5. Notice the installation fails due to incompatible Node.js engines version required.

**Expected Result**

Expect for all Node.js v14 and v12 to be supported, as is indeed the case (you can confirm here https://github.com/nodejs/Release#release-schedule)

**Actual Result**

Getting an error about incomplete Node.js version when trying to install this package

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `5.10.0` |
| `@typescript-eslint/parser`        | `X.Y.Z` |
| `node`                             | `14.15.1` |

**Comment by bradzacher** [comment]
> Is there a reason for specifically setting a lower cap on v12.22 and v14.17 minor versions?

We sync our engine requirements with ESLint. As they push forward, so do we.
Whilst we might not have an explicit need, we test against newer versions only.

We do not test on older versions of node, so we provide no support for older versions.
If it works - it works, if it doesn't - you're on your own. You're free to use `--ignore-engines` or similar to bypass the check in yarn if you wish to use it.

**Comment by lirantal** [comment]
Syncing up with ESLint version requirements make a lot of sense, thanks for explaining Brad.

**Comment by earshinov** [comment]
Related: https://github.com/eslint/eslint/pull/15198
